### PR TITLE
New library versioning

### PR DESCRIPTION
**Pull request by Alexsandruss**
Rename dynamic library files and create new chain of links:

1. Linux:
   - Old: libdaal*.so -> libonedal*.so
   - New: libonedal*.so -> libonedal*.so.{major version} -> libonedal*.so.{major version}.{minor version}
2. MacOS:
   - Old: libdaal*.dylib -> libonedal*.dylib
   - New: libonedal*.dylib -> libonedal*.{major version}.dylib -> libonedal*.{major version}.{minor version}.dylib
3. Windows:
   - Old: onedal_dll.lib
   - New: onedal_dll.{major version}.lib

### CODE REVIEW

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/eb1459f2-7598-f1d6-9b63-8cdcd4b723a1

**Comment by Alexsandruss** [comment]
daal4py build fails due to not implemented support of onedal* library name

**Comment by PetrovKP** [comment]
For daal4py I would suggest updating now. So that the CI is green in two places.

File: makefile [hunk 1]
```diff
@@ -267,6 +267,12 @@ mklgpufpk.HEADERS := $(MKLGPUFPKDIR.include)/mkl_dal_sycl.hpp $(MKLGPUFPKDIR.inc
 # Release library names
 #===============================================================================
 
+major_version := 1
```
Comment (PetrovKP) [inline makefile:270]: Need automizer getting from makefile.ver https://github.com/oneapi-src/oneDAL/blob/master/makefile.ver#L18

File: makefile [hunk 2]
```diff
@@ -267,6 +267,12 @@ mklgpufpk.HEADERS := $(MKLGPUFPKDIR.include)/mkl_dal_sycl.hpp $(MKLGPUFPKDIR.inc
...
+minor_version := 0
```
Comment (PetrovKP) [inline makefile:271]: minnor version is 2 for master branch

Comment (napetrov) [reply makefile:271]: I think it should be 1 for master as gold is 1.0

**Comment by Alexsandruss** [comment]
daal4py update: https://github.com/IntelPython/daal4py/pull/354

File: makefile (hunk 2 above)
Comment (Alexsandruss) [reply makefile:271]: major binary version = 1, minor = 0 (for gold), am I right? I will increment minor version for master after merge of this PR.

File: makefile (hunk 1 above)
Comment (Alexsandruss) [reply makefile]: Added

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/eb153a9f-385e-f114-8fae-8cdcd4b723a1

**Comment by Alexsandruss** [comment]
http://intel-ci.intel.com/eb15431a-628b-f1cb-8797-8cdcd4b723a1

**Review by PetrovKP — APPROVED** [review]

**Review by napetrov — CHANGES_REQUESTED** [review]
There are failures in windows examples related to library name change

File: makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Alexsandruss) [reply makefile]: Already done, we agreed only to add macros in header.

**Comment by Alexsandruss** [comment]
Repaired Win examples job: http://intel-ci.intel.com/eb156918-57d2-f155-abc8-8cdcd4b723a1

**Review by napetrov — APPROVED** [review]

**Pull request merged**